adapters/repos/db/shard_write_inverted.go

```diff
@@ -41,12 +46,20 @@ func (s *Shard) analyzeObject(object *storobj.Object) ([]inverted.Property, []st
 
 	// add nil for all properties that are not part of the object so that they can be added to the inverted index for
 	// the null state (if enabled)
-	var nilProps []string
+	var nilProps []nilProp
 	if s.index.invertedIndexConfig.IndexNullState {
 		for _, prop := range c.Properties {
+			dt := schema.DataType(prop.DataType[0])
+			// some datatypes are not added to the inverted index, so we can skip them here
+			if dt == schema.DataTypeGeoCoordinates || dt == schema.DataTypePhoneNumber || dt == schema.DataTypeBlob {
+				continue
+			}
 			_, ok := schemaMap[prop.Name]
 			if !ok {
-				nilProps = append(nilProps, prop.Name)
+				nilProps = append(nilProps, nilProp{
+					Name:                prop.Name,
+					AddToPropertyLength: !(dt == schema.DataTypeInt || dt == schema.DataTypeNumber || dt == schema.DataTypeBoolean || dt == schema.DataTypeDate),
```

Comment:
Maybe it would be better to create a separate method for checking this like:
```
func isPropertyLength(dt DataType) bool {
switch dt {
case: ...
return false
default:
return true
}
}
```

Reply:
good idea, done